### PR TITLE
chore(daemon): use WaitForCacheSync after sharedInformerFactory Start in unit test

### LINKED ISSUE #126256: chore: use WaitForCacheSync method after sharedInformerFactory Start in integration or unit test follow-ups
### What would you like to be added?

In unit tests and integration tests, after the informer calls the Start method, many files do not call the WaitForCacheSync method to wait for the informer to synchronize.
We should use `WaitForCacheSync` method after sharedInformerFactory `Start`

example: (I just gave examples, not all)

- https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/podautoscaler/horizontal_test.go#L809

- https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/services/namespace_controller.go#L85

- https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/resourcequota/resource_quota_controller_test.go#L134

### Why is this needed?

We should use `WaitForCacheSync` method after sharedInformerFactory `Start`

Because it involves multiple sigs groups, PRs need to be submitted separately, so this issue is used for tracking.

refer: 
https://github.com/kubernetes/kubernetes/pull/126228
https://github.com/kubernetes/kubernetes/pull/126244

### PR DESCRIPTION
**Pull request by omerap12**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Use WaitForCacheSync method after sharedInformerFactory Start
#### Which issue(s) this PR fixes:

Fixes #126256 

#### Special notes for your reviewer:
As requested in issue #126256, I'm splitting this into separate PRs.
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: pkg/controller/daemon/daemon_controller_test.go [hunk 1]
```diff
@@ -553,6 +553,7 @@ func TestExpectationsOnRecreate(t *testing.T) {
 	}
 
 	f.Start(ctx.Done())
+	f.WaitForCacheSync(ctx.Done())
```
Comment (sttts) [inline pkg/controller/daemon/daemon_controller_test.go:556]: Actually, for this one it is done one line below.

Comment (omerap12) [reply pkg/controller/daemon/daemon_controller_test.go:556]: oh right didn't notice. thanks! closing the PR then

**Pull request closed without merging**